git_machete/tests/functional/test_machete.py

```diff
@@ -204,14 +85,311 @@ def launch_command(*args: str) -> str:
+            .check_out("slide_root")
+            .new_branch("child_a")
+            .commit("child_a_1")
+            .push()
+            .check_out("slide_root")
+            .new_branch("child_b")
+            .commit("child_b_1")
+            .push()
+            .check_out("child_b")
+            .new_branch("child_c")
+            .commit("child_c_1")
+            .push()
+            .new_branch("child_d")
+            .commit("child_d_1")
+            .push()
+        )
+
+        self.launch_command("discover", "-y", "--roots=develop")
+
+        self.assertEqual(
+            self.launch_command("status", "-l"),
+            self.adapt(
+                """
+                develop
+                |
+                | slide_root_1
+                o-slide_root
+                  |
+                  | child_a_1
+                  o-child_a
+                  |
+                  | child_b_1
+                  o-child_b
+                    |
+                    | child_c_1
+                    o-child_c
+                      |
+                      | child_d_1
+                      o-child_d *
+                """
+            ),
+        )
+
+        # Slide-out a single interior branch with one downstream. (child_c)
+        # This rebases the single downstream onto the new upstream. (child_b -> child_d)
+
+        self.launch_command("go", "up")
+        self.launch_command("slide-out", "-n")
+
+        self.assertEqual(
+            self.launch_command("status", "-l"),
+            self.adapt(
+                """
+                develop
+                |
+                | slide_root_1
+                o-slide_root
+                  |
+                  | child_a_1
+                  o-child_a
+                  |
+                  | child_b_1
+                  o-child_b
+                    |
+                    | child_d_1
+                    o-child_d * (diverged from origin)
+                """
+            ),
+        )
+
+        # Slide-out an interior branch with multiple downstreams. (slide_root)
+        # This rebases all the downstreams onto the new upstream. (develop -> [child_a, child_b])
+        self.launch_command("traverse", "-Wy")
+        self.launch_command("go", "up")
+        self.launch_command("go", "up")
+
+        self.assertEqual(
+            self.launch_command("status", "-l"),
+            self.adapt(
+                """
+                develop
+                |
+                | slide_root_1
+                o-slide_root *
+                  |
+                  | child_a_1
+                  o-child_a
+                  |
+                  | child_b_1
+                  o-child_b
+                    |
+                    | child_d_1
+                    o-child_d
+                """
+            ),
+        )
+
+        self.launch_command("slide-out", "-n")
+
+        self.assertEqual(
+            self.launch_command("status", "-l"),
+            self.adapt(
```

Comment:
Actually! AFAICS this `self.assertEqual(...., self.adapt(...))` recurs a few time across this test suite... maybe it's worth extracting to a separate method while you're here so that multiple `self.adapt` calls aren't needed?